sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/StorageException.java

```diff
@@ -19,34 +20,29 @@
  */
 public final class StorageException extends HttpResponseException {
     private static final String ERROR_CODE = "x-ms-error-code";
-
-    private final StorageErrorCode errorCode;
-    private final String message;
-
     /**
      * Constructs a {@code StorageException} from the given {@link StorageErrorException}.
      *
-     * @param e The StorageErrorException returned from the service.
-     * @param responseBody The exception body.
+     * @param message the exception message or the response content if a message is not available.
+     * @param response the HTTP response.
+     * @param value the error code of the exception.
      */
-    public StorageException(StorageErrorException e, String responseBody) {
-        super(e.getMessage(), e.getResponse(), e);
-        this.errorCode = StorageErrorCode.fromString(e.getResponse().getHeaders().getValue(ERROR_CODE));
-        this.message = responseBody;
+    public StorageException(String message, HttpResponse response, Object value) {
+        super(message, response, value);
     }
 
     /**
      * @return The error code returned by the service.
      */
     public StorageErrorCode getErrorCode() {
-        return this.errorCode;
+        return StorageErrorCode.fromString(super.getResponse().getHeaders().getValue(ERROR_CODE));
```

Comment:
Why did we remove this from the constructor? We are forcing a parsing of the response headers each time it is accessed.

Reply:
The rest deserilizer in restProxy had a hard time to deserialize the statement in constructor.